optimum/habana/transformers/generation/utils.py

```diff
@@ -952,6 +954,10 @@ def generate(
             input_ids_length=input_ids_length,
             has_token_idx="token_idx" in model_kwargs,
         )
+        if model_input_name == "inputs_embeds" and "token_idx" in model_kwargs:
+            # Need to make sure that generation_config.max_length is correct in this code path
+            # otherwise it will crash withs self._validate_generated_length() below
+            assert input_ids.numel() > 0, "optimum-habana does not support inputs_embeds yet"
```

Comment:
The issue here is that `inputs_embeds` are not supported with static shapes right?

Reply:
correct. Reading it now, I think a `ValueError` error would make more sense. The fact that `input_ids` at this point empty it's just a by product of the two conditions of the `if`.